### PR TITLE
Raise AttributeError in lightning_getattr and lightning_setattr when attribute not found

### LINKED ISSUE #5932: `lightning_getattr` should raise `AttributeError`
## 🚀 ~Feature~ Enhancement
Currently, `lightning_getattr(model, attribute)` raises one of `(ValueError, KeyError, AttributeError` when `attribute` is not found in `model`, but I think it should raise `AttributeError` in all cases since Python built-in `getattr()` only raises `AttributeError`.

https://github.com/PyTorchLightning/pytorch-lightning/blob/4bdf2fe55f45c4cc8b397d4b45041265c402519f/pytorch_lightning/utilities/parsing.py#L241

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #5932.

### Description of the changes
1. Make `lightning_getattr` and `lightning_setattr` raise `AttributeError` but not `ValueError`.
2. Add death tests for `AttributeError`.
2. Rename `lightning_get_*` to `_lightning_get_*` because they are supposed be used within the file.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/utilities/test_parsing.py [hunk 1]
```diff
@@ -107,9 +111,13 @@ def test_lightning_getattr(tmpdir):
     assert lightning_getattr(model7, 'batch_size') == 8, \
         'batch_size not correctly extracted'
 
+    for m in models:
+        with pytest.raises(AttributeError):
```
Comment (rohitgr7) [inline tests/utilities/test_parsing.py:115]: can we have the match string here?

File: tests/utilities/test_parsing.py [hunk 2]
```diff
@@ -126,3 +134,7 @@ def test_lightning_setattr(tmpdir):
         'batch_size not correctly set'
     assert lightning_getattr(model7, 'batch_size') == 128, \
         'batch_size not correctly set'
+
+    for m in models:
+        with pytest.raises(AttributeError):
```
Comment (rohitgr7) [inline tests/utilities/test_parsing.py:139]: can we have the match string here?

File: pytorch_lightning/utilities/parsing.py [hunk 3]
```diff
@@ -254,13 +265,19 @@ def lightning_getattr(model, attribute):
...
     """
-    holders = lightning_get_all_attr_holders(model, attribute)
+    Special setattr for lightning. Checks for attribute in model namespace
+    and the old hparams namespace/dict.
+    Will also set the attribute on datamodule, if it exists.
+
+    Raises:
+        AttributeError:
+            If ``model`` doesn't have ``attribute`` in any of
+            model namespace, the hparams namespace/dict, and the datamodule.
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/parsing.py:276]: +1 for this. shall we add this in other places too. would be good to have the possible exceptions in the docs visible to the user.

**Review by carmocca — APPROVED** [review]
LGTM barring @rohitgr7's comments

**Review by tchaton — APPROVED** [review]
LGTM ! Just need to address @rohitgr7 comments.

**Comment by Borda** [comment]
@akihironitta mind check the comments from @rohitgr7

**Review by Borda — CHANGES_REQUESTED** [review]
add matching text https://github.com/PyTorchLightning/pytorch-lightning/pull/6024/files#r577567364

File: pytorch_lightning/utilities/parsing.py (hunk 3 above)
Comment (akihironitta) [reply pytorch_lightning/utilities/parsing.py:276]: The tracking issue is here: #5434

**Comment by akihironitta** [comment]
@Borda @rohitgr7 The `match` string is added to the tests now, so would you mind having a look at the changes again?

File: pytorch_lightning/utilities/parsing.py [hunk 4]
```diff
@@ -196,9 +196,11 @@ def __repr__(self):
         return out
 
 
-def lightning_get_all_attr_holders(model, attribute):
-    """ Special attribute finding for lightning.  Gets all of the objects or dicts that holds attribute.
-            Checks for attribute in model namespace, the old hparams namespace/dict, and the datamodule. """
+def _lightning_get_all_attr_holders(model, attribute):
+    """
+    Special attribute finding for lightning. Gets all of the objects or dicts that holds attribute.
+    Checks for attribute in model namespace, the old hparams namespace/dict, and the datamodule.
```
Comment (awaelchli) [inline pytorch_lightning/utilities/parsing.py:202]:
```suggestion
    Special attribute finding for Lightning. Gets all of the objects or dicts that holds attribute.
    Checks for attribute in model namespace, the old hparams namespace/dict, and the datamodule.
```
for consistency with the rest of the docs

**Review by awaelchli — APPROVED** [review]
👍 cool

File: pytorch_lightning/utilities/parsing.py (hunk 4 above)
Comment (akihironitta) [reply pytorch_lightning/utilities/parsing.py]: Updated!

File: pytorch_lightning/utilities/parsing.py (hunk 3 above)
Comment (rohitgr7) [reply pytorch_lightning/utilities/parsing.py:276]: nicE!

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**